### PR TITLE
Polish async suggestions

### PR DESCRIPTION
**Pull request by DoctorKrolic**
Follow up to https://github.com/dotnet/roslyn/pull/60528

1) Do not supress snippets - user can have a type shortcut, e.g. `vtask` for `ValueTask`
2) Show type if it contains valid task-like nested types
3) Rename `IsInTaskLikeTypeContext` -> `IsTaskLikeTypeContext` to keep things consistent

@CyrusNajmabadi Would like you to look through

### CODE REVIEW

File: src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs [hunk 1]
```diff
@@ -92,7 +92,8 @@ private static bool IsValidForTaskLikeTypeOnlyContext(ISymbol symbol, TSyntaxCon
                        namedType.Equals(compilation.IAsyncEnumeratorOfTType());
             }
 
-            return symbol.IsAwaitableNonDynamic(context.SemanticModel, context.Position);
+            return namedType.IsAwaitableNonDynamic(context.SemanticModel, context.Position) ||
+                   namedType.GetTypeMembers().Any(m => IsValidForTaskLikeTypeOnlyContext(m, context));
```
Comment (sharwell) [inline src/Features/Core/Portable/Completion/Providers/AbstractRecommendationServiceBasedCompletionProvider.cs:96]:
```suggestion
                   namedType.GetTypeMembers().Any(static (m, context) => IsValidForTaskLikeTypeOnlyContext(m, context), context);
```

**Review by sharwell — COMMENTED** [review]
Seems fine overall. @CyrusNajmabadi to confirm behavior changes

**Review by CyrusNajmabadi — APPROVED** [review]

**Comment by CyrusNajmabadi** [comment]
Thanks!

**Comment by DoctorKrolic** [comment]
@CyrusNajmabadi Restart failing CI pls

**Pull request merged**